### PR TITLE
[4.5] cgroup, systemd: cleanup cgroups

### PR DESCRIPTION
**Pull request by haircommander**
some hierarchies were created directly by .Apply() on top of systemd
managed cgroups.  systemd doesn't manage these and as a result we leak
these cgroups.

Cherry-picked-by: Peter Hunt <pehunt@redhat.com>
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

[This is a backport of opencontainers/runc#2503, fixing a regression introduced by commit bfa1b2a]

### CODE REVIEW

**Comment by haircommander** [comment]
should fix https://bugzilla.redhat.com/show_bug.cgi?id=1880876

**Comment by haircommander** [comment]
ptal @mrunalp @kolyshkin

**Comment by haircommander** [comment]
also added https://github.com/opencontainers/runc/commit/254d23b964f1f32ccd93bca1c3e25c173e75ed7c

**Comment by mrunalp** [comment]
Looks fine. I will wait for @kolyshkin to do final review and merge.

**Comment by kolyshkin** [comment]
This is a backport of https://github.com/opencontainers/runc/pull/2503, fixing a regression introduced by commit bfa1b2aab39f7735db9319478b8f63b9adca26e3

**Review by kolyshkin — APPROVED** [review]
LGTM

**Pull request merged**
